lambdas/service/app.py

```diff
@@ -655,204 +655,233 @@ def repository_search(entity_type: str, item_id: str):
         raise NotFoundError(msg=e)
 
 
-@app.route('/repository/files', methods=['HEAD', 'GET'], cors=True)
-@app.route('/repository/files/{file_id}', methods=['GET'], cors=True)
+generic_object_spec = schema.object(additional_properties=True)
+array_of_object_spec = schema.array(generic_object_spec)
+hit_spec = schema.object(
+    additional_properties=True,
+    protocols=array_of_object_spec,
+    entryId=str,
+    samples=array_of_object_spec,
+    specimens=array_of_object_spec,
+    cellLines=array_of_object_spec,
+    donorOrganisms=array_of_object_spec,
+    organoids=schema.array(str),
+    cellSuspensions=array_of_object_spec,
+)
+
+page_spec = schema.object(
+    hits=schema.array(hit_spec),
+    pagination=generic_object_spec,
+    termFacets=generic_object_spec
+)
+
+
+def filters_param_spec(facets):
+    return params.query(
+        'filters',
+        schema.optional(
+            application_json(
+                schema.object(**{
+                    facet: schema.optional(
+                        schema.object(**{
+                            relation: schema.optional(schema.array({}))
+                            for relation in ('is', 'contains', 'within', 'intersects')
+                        })
+                    ) for facet in facets
```

Comment:
This results in a very verbose example. I think you should hard-code a more typical one.